src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -0,0 +1,307 @@
+
+            Assert.Equal(count, queue.Count);
+            queue.Clear();
+            Assert.Equal(expected: 0, queue.Count);
+        }
+
+        #endregion
+
+        #region EnsureCapacity, TrimExcess
+
+        [Theory]
+        [InlineData(0, 0)]
+        [InlineData(0, 5)]
+        [InlineData(1, 1)]
+        [InlineData(3, 100)]
+        public void PriorityQueue_Generic_TrimExcess_ValidQueueThatHasntBeenRemovedFrom(int initialCapacity, int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity, count, out _);
+            queue.TrimExcess();
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_TrimExcess_Repeatedly(int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity: 0, count, out _);
+
+            Assert.Equal(count, queue.Count);
+            queue.TrimExcess();
+            queue.TrimExcess();
+            queue.TrimExcess();
+            Assert.Equal(count, queue.Count);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidPositiveCollectionSizes))]
+        public void PriorityQueue_Generic_EnsureCapacityAndTrimExcess(int count)
+        {
+            IReadOnlyCollection<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToArray();
+            PriorityQueue<TElement, TPriority> queue = new PriorityQueue<TElement, TPriority>();
+            int expectedCount = 0;
+            Random random = new Random(Seed: 34);
+            int getNextEnsureCapacity() => random.Next(0, count * 2);
+            void trimAndEnsureCapacity()
+            {
+                queue.TrimExcess();
+                queue.EnsureCapacity(getNextEnsureCapacity());
+                queue.TrimExcess();
+            };
+
+            foreach (var (element, priority) in itemsToEnqueue)
+            {
+                trimAndEnsureCapacity();
+                queue.Enqueue(element, priority);
+                expectedCount++;
+                Assert.Equal(expectedCount, queue.Count);
+            }
+
+            while (expectedCount > 0)
+            {
+                queue.Dequeue();
+                trimAndEnsureCapacity();
+                expectedCount--;
+                Assert.Equal(expectedCount, queue.Count);
+            }
+
+            trimAndEnsureCapacity();
+            Assert.Equal(0, queue.Count);
+        }
+
+        #endregion
+
+        #region Enumeration ordering
+
+        [Theory]
+        [MemberData(nameof(ValidPositiveCollectionSizes))]
+        public void PriorityQueue_EnumerationIsConsistent(int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity: 0, count, out _);
+
+            (TElement, TPriority)[] firstEnumeration = queue.UnorderedItems.ToArray();
```

Comment:
Consider adding a test that checks for enumerator invalidation when the collection is modified.

Reply:
Covered in `PriorityQueue_Enumeration_InvalidationOnModifiedCollection` and `PriorityQueue_Enumeration_InvalidationOnModifiedCapacity`.